File: examples/agent-card-showcase/data-analysis-agent.js
import { phlowAuth } from 'phlow-auth';

// Define your agent's capabilities with an Agent Card
const dataAnalysisAgent = phlowAuth({
  agentCard: {
    name: "DataWizard",
    description: "AI agent specialized in data analysis and visualization",
    skills: ["data-analysis", "visualization", "statistical-modeling"],
    endpoints: {
      analyze: { method: "POST", path: "/analyze" },
      visualize: { method: "POST", path: "/visualize" }
    }
  }
});

// Your agent's endpoints - protected automatically
app.post('/analyze', dataAnalysisAgent.authenticate(), async (req, res) => {
  const { dataset, analysis_type } = req.body;
  // Perform analysis...
  res.json({ results: "Analysis complete", requestedBy: req.agent.name });
});

// Agent card automatically exposed at /.well-known/agent.json
app.get('/.well-known/agent.json', dataAnalysisAgent.wellKnownHandler());

File: examples/agent-card-showcase/data-analysis-agent.py
from phlow_auth import PhlowAuth
from flask import Flask, request, jsonify

app = Flask(__name__)

# Define your agent's capabilities with an Agent Card
phlow = PhlowAuth(
    agent_card={
        "name": "DataWizard",
        "description": "AI agent specialized in data analysis and visualization",
        "skills": ["data-analysis", "visualization", "statistical-modeling"],
        "endpoints": {
            "analyze": {"method": "POST", "path": "/analyze"},
            "visualize": {"method": "POST", "path": "/visualize"}
        }
    }
)

# Your agent's endpoints - protected automatically
@app.route('/analyze', methods=['POST'])
@phlow.authenticate()
def analyze():
    data = request.json
    dataset = data.get('dataset')
    analysis_type = data.get('analysis_type')
    # Perform analysis...
    return jsonify({
        "results": "Analysis complete",
        "requestedBy": request.agent['name']
    })

# Agent card automatically exposed
@app.route('/.well-known/agent.json')
def agent_card():
    return phlow.well_known_handler()